src/frontend/src/lib/derived/address.derived.ts

```diff
@@ -1,19 +1,24 @@
-import { addressStore } from '$lib/stores/address.store';
+import { ETHEREUM_TOKEN_ID } from '$env/tokens.env';
+import { addressStore, type AddressData } from '$lib/stores/address.store';
 import type { OptionEthAddress } from '$lib/types/address';
-import { isNullish } from '@dfinity/utils';
+import { isNullish, nonNullish } from '@dfinity/utils';
 import { derived, type Readable } from 'svelte/store';
 
 export const addressNotLoaded: Readable<boolean> = derived([addressStore], ([$addressStore]) =>
 	isNullish($addressStore)
 );
 
 export const ethAddress: Readable<OptionEthAddress> = derived([addressStore], ([$addressStore]) =>
-	$addressStore === null ? null : $addressStore?.address
+	$addressStore?.[ETHEREUM_TOKEN_ID] === null ? null : $addressStore?.[ETHEREUM_TOKEN_ID].data
 );
 
 export const addressCertified: Readable<boolean> = derived(
 	[addressStore],
-	([$addressStore]) => $addressStore?.certified === true
+	([$addressStore]) =>
+		nonNullish($addressStore) &&
+		Object.values($addressStore).every(
```

Comment:
This introduces a different behavior. I would stick to certfying only the Eth address for now. In another PR we can rename `addressCertified` to `ethAddressCertified`.

Reply:
changed